master/buildbot/data/builds.py

```diff
@@ -143,6 +143,16 @@ def get(self, resultSpec, kwargs):
             buildscol.append(data)
         defer.returnValue(buildscol)
 
+    def startConsuming(self, callback, options, kwargs):
+        builderid = kwargs.get('builderid')
+        if builderid is not None:
+            return self.master.mq.startConsuming(callback,
+                                                 ('builders', str(builderid),
+                                                  'builds', None, None))
+        else:
+            return self.master.mq.startConsuming(callback,
```

Comment:
it's a personal preference, but I'd rather calculate the routing key and then use _one_ call to `self.master.mq.startConsuming`